x-pack/plugin/core/src/test/java/org/elasticsearch/protocol/xpack/XPackInfoResponseTests.java

```diff
@@ -33,6 +35,34 @@ protected XPackInfoResponse createBlankInstance() {
         return new XPackInfoResponse();
     }
 
+    @Override
+    public org.elasticsearch.client.xpack.XPackInfoResponse doHLRCParseInstance(XContentParser parser) throws IOException {
+        return org.elasticsearch.client.xpack.XPackInfoResponse.fromXContent(parser);
+    }
+
+    @Override
+    public XPackInfoResponse convert(org.elasticsearch.client.xpack.XPackInfoResponse instance) {
+        final org.elasticsearch.client.xpack.XPackInfoResponse.BuildInfo buildInfo = instance.getBuildInfo();
+        final XPackInfoResponse.BuildInfo buildInfo1 =
+            buildInfo != null ? new BuildInfo(buildInfo.getHash(), buildInfo.getTimestamp()) : null;
+        final org.elasticsearch.client.xpack.XPackInfoResponse.LicenseInfo licenseInfo = instance.getLicenseInfo();
+        final XPackInfoResponse.LicenseInfo licenseInfo1 =
+            licenseInfo != null
+            ? new LicenseInfo(licenseInfo.getUid(), licenseInfo.getType(), licenseInfo.getMode(),
+                licenseInfo.getStatus() != null ? LicenseStatus.valueOf(licenseInfo.getStatus().name()) : null,
+                licenseInfo.getExpiryDate())
+            : null;
+        final org.elasticsearch.client.xpack.XPackInfoResponse.FeatureSetsInfo featureSetsInfo = instance.getFeatureSetsInfo();
+        final XPackInfoResponse.FeatureSetsInfo featureSetsInfo1 =
+            featureSetsInfo != null
+            ? new FeatureSetsInfo(featureSetsInfo.getFeatureSets().values().stream()
+               .map(fs -> new XPackInfoResponse.FeatureSetsInfo.FeatureSet(fs.name(), fs.description(), fs.available(), fs.enabled(),
+                   fs.nativeCodeInfo()))
+               .collect(Collectors.toSet()))
+            : null;
+        return new XPackInfoResponse(buildInfo1, licenseInfo1, featureSetsInfo1);
```

Comment:
I think it'd be cleaner to make a few "private convertHlrcToInternal` methods each with different input and output types. It might not be. I dunno. all of the fully qualified names make this look bigger than it really is.

Reply:
done in b5372c4